aspnetcore/web-api/action-return-types.md

```diff
@@ -33,14 +33,7 @@ When multiple return types are possible, it's common to mix an <xref:Microsoft.A
 
 ### Return IEnumerable\<T> or IAsyncEnumerable\<T>
 
-Returning <xref:System.Collections.Generic.IEnumerable%601> from an action results in synchronous collection iteration by the serializer. The result is the blocking of calls and a potential for thread pool starvation. To illustrate, imagine that Entity Framework (EF) Core is being used for the web API's data access needs. The following action's return type is synchronously enumerated during serialization:
-
-```csharp
-public IEnumerable<Product> GetOnSaleProducts() =>
-    _context.Products.Where(p => p.IsOnSale);
-```
-
-Consider declaring the action signature's return type as `IAsyncEnumerable<T>` to guarantee the asynchronous iteration. Ultimately, the iteration mode is based on the underlying concrete type being returned. MVC automatically buffers any concrete type that implements `IAsyncEnumerable<T>`.
+ASP.NET Core buffers the result of actions that return <xref:System.Collections.Generic.IEnumerable%601> before writing them to the response. Consider declaring the action signature's return type as <xref:System.Collections.Generic.IAsyncEnumerable%601> to guarantee asynchronous iteration. Ultimately, the iteration mode is based on the underlying concrete type being returned. MVC automatically buffers any concrete type that implements `IAsyncEnumerable<T>`.
```

Comment:
I _think_ this covers the 3.0+ stuff. The stuff I removed was prefixed with "In ASP.NET Core 2.2 and earlier" before, so I also dropped the sample code as the exact same code sample is shown right after now.